src/protocols/bdx/AsyncTransferFacilitator.h

```diff
@@ -72,7 +72,11 @@ class AsyncTransferFacilitator : public Messaging::ExchangeDelegate
      */
     virtual void DestroySelf() = 0;
 
-    // Calling ProcessOutputEvents can destroy this object before the call returns.
+    /**
+     * Calling ProcessOutputEvents can destroy this object before the call returns for certain cases
+     * where the state machine either receives a message indicating termination of the transfer session (StatusReport
+     * or BlockAckEOF) or generates an InternalError for various error scenarios including timeout.
```

Comment:
The exact cases when it can happen are not really part of the public API, so why do we want to describe them here?